### PR TITLE
Add Emscripten core tests

### PR DESCRIPTION
**Pull request by bjfish**
- Updated readme and attributions
- Updated build.rs to glob for c/cpp sources in emtests
- Updated build.rs to disable cc and out generation by default (txt/out) files provided
- Added an `ignores.txt` file to add `#[ignore]` tag to these tests
- Added emscripten core tests with `ignores.txt` listing for failing tests

### CODE REVIEW

**Comment by syrusakbary** [comment]
The PR looks great.

I'm not sure why it includes so many `.txt` files (are they really used?).
But once that question is resolved we should be good to merge 👍

**Comment by xmclark** [comment]
This looks cool! I have a couple questions.

What is the rationale for not compiling c files? I can see a few benefits, but I was wondering if you could summarize the reasoning for changing this. Overall, I like the idea. This caused me some confusion before.

Do we want to separate the emscripten tests from other sys call related tests? For example: there are other sys calls that we test for that are not apart of the core emscripten tests. I'm thinking of using a directory for the core emscripten test.

**Comment by bjfish** [comment]
@syrusakbary It appears both `.out` and `.txt` are used for output files: https://github.com/kripken/emscripten/blob/942b9d07e7cf2ac6a92e96d0fa6f60df8bb32456/tests/test_core.py#L111

@xmclark 
> What is the rationale for not compiling c files?
In the case of Emscripten, all the output has already been generated and checked in. So, compiling and running these seemed redundant.

> Do we want to separate the emscripten tests from other sys call related tests?
Yes, I think we should work towards a clean organization of tests into emscripten/non-emscripten test.  I chose this organization for now because it was the most straightforward to implement and the `emtests` directory appeared to be the best fit for these.

Later on I hope this would be organized something like this for example:
```
tests/
tests/syscalls/
tests/spectests
tests/emtests/core
```

File: build/emtests.rs [hunk 1]
```diff
@@ -4,41 +4,43 @@
...
-        .output()
-        .expect("failed to execute process");
+    //    let output = Command::new(output_str)
+    //        .arg(output_str)
+    //        .output()
+    //        .expect("failed to execute process");
 
     // Remove executable
-    fs::remove_file(output_str).unwrap();
+    //    fs::remove_file(output_str).unwrap();
```
Comment (xmclark) [inline build/emtests.rs]: If we no longer run `cc`, why not do the same with the `emcc`?

Comment (bjfish) [reply build/emtests.rs]: The emscripten tests don't include generated wasm files like they do the output. Also, we may need to updated the wasm files as EMCC is updated.  If you're just want to regenerate test files though it much faster to comment out EMCC temporarily. Maybe it should be commented out by default.

File: build/emtests.rs [hunk 2]
```diff
@@ -98,10 +137,26 @@ pub fn build() {
...
     let mut modules: Vec<String> = Vec::new();
     // modules.reserve_exact(TESTS.len());
-    for test in TESTS.iter() {
-        let moudle_name = compile(test);
-        modules.push(format!("mod {};", moudle_name));
+
+    let ignores = read_ignore_list();
+
+    for ext in EXTENSIONS.iter() {
+        for entry in glob(&format!("emtests/*.{}", ext)).unwrap() {
```
Comment (syrusakbary) [inline build/emtests.rs]: Is it possible to remove the double loop to do something like this instead?

```rust
        for entry in glob("emtests/*.(c|cpp)").unwrap() {
```

Comment (bjfish) [reply build/emtests.rs]: I've tried that and `{c,cpp}`. I don't believe this is supported yet for the glob crate:
https://github.com/rust-lang-nursery/glob/blob/master/tests/glob-std.rs

Comment (syrusakbary) [reply build/emtests.rs]: This one perhaps?
```
        for entry in glob("emtests/*.cp?p?)").unwrap() {
```

However, it opens the possibility of getting unwanted `abc.cp` files

Comment (syrusakbary) [reply build/emtests.rs]: And not that clean. maybe we can keep the double loop for now.

**Pull request merged**